### PR TITLE
#652[Feature request] Provide compartments as objects

### PR DESCRIPTION
**Pull request by MaxGreil**
#652 
Implemented model.compartments as objects, therefore added new class compartment in cobra.core.
- In cobra.core: modified class cobra.model to store new class compartment correctly.
- In cobra.io: modified json.py, dict.py, mat.py, sbml.py and sbml3.py.
- In cobra.manipulation: modified modify.py.
- In cobra.test.data: run update_pickles.py and modified existing data files according to new class compartment.
- In cobra.test: modified tests according to new class compartment.

EDIT 20.04.:
- In cobra.util: modified version_info.py by importing pkg_resources from setuptools for function pkg_resources.working_set. Dropped pip.get_installed_distributions().

EDIT 21.04.:
- In cobra.test: modified test_io.py to be compartible for pickles from both python2 and python3

### CODE REVIEW

**Comment by MaxGreil** [comment]
All TravisCI tests are now working.
The new funcionality of the PR is as requested in #652 . Compartments are now not just a dictionary but objects like all the other model components. They can be accessed like all other model components with `model.compartments` which returns the DictList containing all compartments of the model.
New class `compartment` inherits from class `object` and therefore possesses the same functionality like all the other model components. 
Code reviewer: @cdiener

**Comment by MaxGreil** [comment]
@Midnighter I used the update_pickles.py script.
As far as I can tell, the files like mini_cobra.xml and mini_fbc*.xml were a bit shifted around a bit but this did not change the context/variables/values.
The same goes for the textbook* solutions which have no changed values but were as before shifted around.

File: cobra/test/data/mini.yml [hunk 1]
```diff
@@ -1143,6 +1143,10 @@
     - name: G_s0001
 - id: mini_textbook
 - compartments:
-    e: extracellular
-    c: cytosol
+    - !!omap
```
[the remark this answers is not in the document]
Comment (Midnighter) [reply cobra/test/data/mini.yml]: The ordered map is introduced by the YAML IO automatically but I agree, let's keep the old format of `id: name`.

File: cobra/test/test_io.py [hunk 2]
```diff
@@ -129,7 +129,10 @@ def raise_libsbml_errors():
 def io_trial(request, data_directory):
     with open(join(data_directory, request.param.reference_file),
               "rb") as infile:
-        reference_model = load(infile)
+        try:
+            reference_model = load(infile)
+        except UnicodeDecodeError:
+            reference_model = load(infile, encoding='latin1')
```
[the remark this answers is not in the document]
Comment (Midnighter) [reply cobra/test/test_io.py]: I agree, please store files in utf-8 format.

File: cobra/util/version_info.py [hunk 3]
```diff
@@ -12,6 +12,8 @@
 
 import pkg_resources
 
+import pkg_resources
```
[the remark this answers is not in the document]
Comment (Midnighter) [reply cobra/util/version_info.py]: Please revert the changes to this file since the issue was addressed in #697.

**Comment by cdiener** [comment]
Appveyor seems to work again.

**Comment by Midnighter** [comment]
Dear @MaxGreil, I'm very sorry to be doing this but we need to move things along right now. Thus I took your branch, rebased it on the current devel and will take over development on that new branch. Thank you very much for contribution and I hope this is not offending you. Please see #725 for the new PR.

**Comment by MaxGreil** [comment]
Hi @Midnighter, I didn't have the time to continue working on this issue.

**Comment by Midnighter** [comment]
That's completely understandable. Would you like to finish it up, though?

**Pull request closed without merging**